### PR TITLE
feat: support disable SSH

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -1507,8 +1507,17 @@ type NodeBootstrappingConfiguration struct {
 	IsARM64                        bool
 	CustomCATrustConfig            *CustomCATrustConfig
 	DisableUnattendedUpgrades      bool
+	SSHStatus                      SSHStatus
 }
 
+type SSHStatus int
+
+const (
+	SSHUnspecified SSHStatus = 0
```
Comment (alexeldeib) [inline pkg/agent/datamodel/types.go:1516]: did you want to try iota here?

Comment (UtheMan) [reply pkg/agent/datamodel/types.go]: done

File: pkg/agent/baker_test.go [hunk 2]
```diff
@@ -135,6 +135,7 @@ var _ = Describe("Assert generated customData and cseCmd", func() {
 			PrimaryScaleSetName:           "aks-agent2-36873793-vmss",
 			IsARM64:                       false,
 			DisableUnattendedUpgrades:     false,
+			SSHStatus:                     datamodel.SSHUnspecified,
```
Comment (alexeldeib) [inline pkg/agent/baker_test.go:138]: can you add/modify a unit test case for enable/disable ssh and ensure the generated code looks fine? specifically cse_main

**Review by alexeldeib — APPROVED** [review]

File: pkg/agent/testdata/AKSUbuntu2204+SSHStatusOff/line33.sh [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+wait_for_file 3600 1 /opt/azure/containers/provision_installs.sh || exit $ERR_FILE_WATCH_TIMEOUT
+source /opt/azure/containers/provision_installs.sh
+
+wait_for_file 3600 1 /opt/azure/containers/provision_installs_distro.sh || exit $ERR_FILE_WATCH_TIMEOUT
+source /opt/azure/containers/provision_installs_distro.sh
+
+wait_for_file 3600 1 /opt/azure/containers/provision_configs.sh || exit $ERR_FILE_WATCH_TIMEOUT
+source /opt/azure/containers/provision_configs.sh
+disableSSH || exit $ERR_DISABLE_SSH
```
Comment (alexeldeib) [inline pkg/agent/testdata/AKSUbuntu2204+SSHStatusOff/line33.sh:58]: correct: SSHStatusOff disables SSH

File: pkg/agent/testdata/AKSUbuntu2204+SSHStatusOn/line33.sh [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+source /opt/azure/containers/provision_source_distro.sh
+
+wait_for_file 3600 1 /opt/azure/containers/provision_installs.sh || exit $ERR_FILE_WATCH_TIMEOUT
+source /opt/azure/containers/provision_installs.sh
+
+wait_for_file 3600 1 /opt/azure/containers/provision_installs_distro.sh || exit $ERR_FILE_WATCH_TIMEOUT
+source /opt/azure/containers/provision_installs_distro.sh
+
+wait_for_file 3600 1 /opt/azure/containers/provision_configs.sh || exit $ERR_FILE_WATCH_TIMEOUT
+source /opt/azure/containers/provision_configs.sh
```
Comment (alexeldeib) [inline pkg/agent/testdata/AKSUbuntu2204+SSHStatusOn/line33.sh:57]: correct: disableSSH would be here, but because SSHStatusOn, it's not

**Review by alexeldeib — APPROVED** [review]

**Pull request merged**